include/graphit/frontend/fir.h

```diff
@@ -1654,6 +1656,22 @@ namespace graphit {
                virtual void copy(FIRNode::Ptr);
         };
 
+        struct FuncExpr : public Expr {
```

Comment:
Do we need another FIR node for the FuncExpr parameter? Since all these nodes are expressions, can we not use Call and make the name/target as an expression?